src/builtin.c

```diff
@@ -1201,7 +1201,29 @@ static jv f_string_implode(jq_state *jq, jv a) {
   if (jv_get_kind(a) != JV_KIND_ARRAY) {
     return ret_error(a, jv_string("implode input must be an array"));
   }
-  return jv_string_implode(a);
+
+  int len = jv_array_length(jv_copy(a));
+  assert(len >= 0);
```

Comment:
@nicowilliams wops, this assert ok/usefull?

Reply:
Indeed, it's not!

Reply:
Fixed